network/p2p/p2p.go

```diff
@@ -164,9 +167,53 @@ func (n *p2pNetwork) Start(logger *zap.Logger) error {
 		return nil
 	}
 
-	logger.Info("starting")
+	connector := make(chan peer.AddrInfo, connectorQueueSize)
+	go func() {
+		// Wait for own subnets to be subscribed to and updated.
+		// TODO: wait more intelligently with a channel.
+		time.Sleep(8 * time.Second)
```

Comment:
what if we are freshly syncing? this might take much longer

Reply:
@y0sher  IIRC we should be starting p2p after we've finished syncing history

Reply:
but subscribing to subnet might happen later after p2p already started. for example when downloading metadata